buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -64,6 +65,17 @@ public void apply(Project project) {
         project.getPluginManager().apply(RestTestBasePlugin.class);
         project.getPluginManager().apply(RestResourcesPlugin.class);
         project.getPluginManager().apply(YamlRestTestPlugin.class);
+        project.getPluginManager().apply(CheckRestCompatPlugin.class);
+
+        final Consumer<Task> enabledFunction = task -> {
+            // marry the enablement of these tasks to the enablement of the lifecycle task
```

Comment:
No please let's not do this. This opens up the door for all kinds of issues due to configuration ordering. We should skip these tasks at _execution_ time via an `onlyIf` condition. All tasks that do work (the tests) should simply have an `onlyIf` condition on them [as we do with all other bwc tests](https://github.com/elastic/elasticsearch/blob/90e10615af16f2c0cb182938496d24d500e283f4/gradle/bwc-test.gradle#L24).

There's no need to disable/skip the lifecyle tasks as the don't actually do work.

Reply:
used onlyIf instead of enabled=false, removed the disabling of the lifecycle task, and allow the YamlRestCompatTestPlugin to read the ext directly. https://github.com/elastic/elasticsearch/pull/67394/commits/176a72ed820d97948ab82e7bd736bf9201ef6be6